mnefun/_mnefun.py

```diff
@@ -2405,24 +2411,57 @@ def do_preprocessing_combined(p, subjects, run_indices):
                              pick=True, skip_by_annotation='edge',
                              **fir_kwargs)
             events = fixed_len_events(p, raw)
+            rtmin = p.reject_tmin \
+                if p.reject_tmin is not None else p.tmin
+            rtmax = p.reject_tmax \
+                if p.reject_tmax is not None else p.tmax
             # do not mark eog channels bad
             meg, eeg = 'meg' in raw, 'eeg' in raw
             picks = pick_types(raw.info, meg=meg, eeg=eeg, eog=False,
                                exclude=[])
             assert p.auto_bad_flat is None or isinstance(p.auto_bad_flat, dict)
             assert p.auto_bad_reject is None or isinstance(p.auto_bad_reject,
-                                                           dict)
-            if p.auto_bad_reject is None and p.auto_bad_flat is None:
+                                                           dict) or \
+                   p.auto_bad_reject == 'auto'
+            if p.auto_bad_reject is 'auto':
+                print('    Auto bad channel selection active. '
+                      'Will try using Autoreject module to '
+                      'compute rejection criterion.\n' , end='')
+                try:
+                    from autoreject import get_rejection_threshold
+                except ImportError:
+                    raise ImportError('     Autoreject module not installed.\n'
+                                      '     Noisy channel detection parameter '
+                                      '     not defined. To use autobad '
+                                      '     channel selection either define '
+                                      '     rejection criteria or install '
+                                      '     Autoreject module.\n')
+                else:
+                    print('    Computing thresholds.\n', end='')
```

Comment:
No need to nest these lines in an `else` clause anymore (they will only execute if there was no `ImportError` either way)